## Final Project Submission

Please fill out:
* Group members: Myra Kadenge, Edmund Nyaribo ,Frank Oyugi, Silvia Gworit
* Student pace: full time
* Scheduled project review date/time: 
* Instructor name: Maryann Mwikali
* Blog post URL:


## Overview
The King County Housing Data Set provides information about houses in King County, including their size, location, condition, and other features. The objective of this project is to create multiple regression models that can accurately predict the price of a house.

The project is commissioned by a commercial real estate agaency interested in assisting homeowners to buy or sell homes within specific areas.


## Data preparation

In [1]:
import pandas as pd

In [16]:
data=pd.read_csv('data/kc_house_data.csv', index_col=0)
data.head()

date     price  bedrooms  bathrooms  sqft_living  sqft_lot  \
id                                                                             
7129300520  10/13/2014  221900.0         3       1.00         1180      5650   
6414100192   12/9/2014  538000.0         3       2.25         2570      7242   
5631500400   2/25/2015  180000.0         2       1.00          770     10000   
2487200875   12/9/2014  604000.0         4       3.00         1960      5000   
1954400510   2/18/2015  510000.0         3       2.00         1680      8080   

            floors waterfront  view  condition          grade  sqft_above  \
id                                                                          
7129300520     1.0        NaN  NONE    Average      7 Average        1180   
6414100192     2.0         NO  NONE    Average      7 Average        2170   
5631500400     1.0         NO  NONE    Average  6 Low Average         770   
2487200875     1.0         NO  NONE  Very Good      7 Average        1050   
1954400510     1.0         NO  NONE    Average         8 Good        1680   

           sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
id                                                                            
7129300520           0.0      1955           0.0    98178  47.5112 -122.257   
6414100192         400.0      1951        1991.0    98125  47.7210 -122.319   
5631500400           0.0      1933           NaN    98028  47.7379 -122.233   
2487200875         910.0      1965           0.0    98136  47.5208 -122.393   
1954400510           0.0      1987           0.0    98074  47.6168 -122.045   

            sqft_living15  sqft_lot15  
id                                     
7129300520           1340        5650  
6414100192           1690        7639  
5631500400           2720        8062  
2487200875           1360        5000  
1954400510           1800        7503

In [17]:
data = data.sort_values(by='yr_renovated', ascending=False)

In [18]:
data

date      price  bedrooms  bathrooms  sqft_living  sqft_lot  \
id                                                                              
7284900030   5/22/2014   850000.0         4       3.25         3090      6744   
8712100320   7/28/2014   585000.0         5       2.75         2350      4178   
6821100195   3/31/2015   830000.0         4       3.00         2020      6000   
8935100100    7/1/2014   476000.0         4       3.00         2890      6885   
9141100005  10/28/2014   285000.0         4       3.50         2770     10505   
...                ...        ...       ...        ...          ...       ...   
1931300412   4/16/2015   475000.0         3       2.25         1190      1200   
8672200110   3/17/2015  1090000.0         5       3.75         4170      8142   
1972201967  10/31/2014   520000.0         2       2.25         1530       981   
191100405    4/21/2015  1580000.0         4       3.25         3410     10125   
7202300110   9/15/2014   810000.0         4       3.00         3990      7838   

            floors waterfront       view condition         grade  sqft_above  \
id                                                                             
7284900030     2.0         NO  EXCELLENT   Average      9 Better        3090   
8712100320     1.5         NO       NONE   Average        8 Good        1520   
6821100195     1.0         NO       NONE   Average        8 Good        1220   
8935100100     1.0         NO       NONE   Average     7 Average        1590   
9141100005     2.0         NO       NONE   Average        8 Good        2770   
...            ...        ...        ...       ...           ...         ...   
1931300412     3.0         NO       NONE   Average        8 Good        1190   
8672200110     2.0         NO    AVERAGE   Average  10 Very Good        4170   
1972201967     3.0         NO       NONE   Average        8 Good        1480   
191100405      2.0         NO       NONE   Average  10 Very Good        3410   
7202300110     2.0         NO       NONE   Average      9 Better        3990   

           sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
id                                                                            
7284900030           0.0      1923        2015.0    98177  47.7680 -122.388   
8712100320         830.0      1922        2015.0    98112  47.6388 -122.300   
6821100195         800.0      1968        2015.0    98199  47.6563 -122.401   
8935100100        1300.0      1945        2015.0    98115  47.6763 -122.282   
9141100005           0.0      1940        2015.0    98133  47.7412 -122.355   
...                  ...       ...           ...      ...      ...      ...   
1931300412           0.0      2008           NaN    98103  47.6542 -122.346   
8672200110           0.0      2006           NaN    98056  47.5354 -122.181   
1972201967          50.0      2006           NaN    98103  47.6533 -122.346   
191100405              ?      2007           NaN    98040  47.5653 -122.223   
7202300110           0.0      2003           NaN    98053  47.6857 -122.046   

            sqft_living15  sqft_lot15  
id                                     
7284900030           2020        6656  
8712100320           1920        4178  
6821100195           1400        6000  
8935100100           2180        6885  
9141100005           1760       10505  
...                   ...         ...  
1931300412           1180        1224  
8672200110           3030        7980  
1972201967           1530        1282  
191100405            2290       10125  
7202300110           3370        6814  

[21597 rows x 20 columns]

In [12]:
data.describe()

price      bedrooms     bathrooms   sqft_living      sqft_lot  \
count  2.159700e+04  21597.000000  21597.000000  21597.000000  2.159700e+04   
mean   5.402966e+05      3.373200      2.115826   2080.321850  1.509941e+04   
std    3.673681e+05      0.926299      0.768984    918.106125  4.141264e+04   
min    7.800000e+04      1.000000      0.500000    370.000000  5.200000e+02   
25%    3.220000e+05      3.000000      1.750000   1430.000000  5.040000e+03   
50%    4.500000e+05      3.000000      2.250000   1910.000000  7.618000e+03   
75%    6.450000e+05      4.000000      2.500000   2550.000000  1.068500e+04   
max    7.700000e+06     33.000000      8.000000  13540.000000  1.651359e+06   

             floors    sqft_above      yr_built  yr_renovated       zipcode  \
count  21597.000000  21597.000000  21597.000000  17755.000000  21597.000000   
mean       1.494096   1788.596842   1970.999676     83.636778  98077.951845   
std        0.539683    827.759761     29.375234    399.946414     53.513072   
min        1.000000    370.000000   1900.000000      0.000000  98001.000000   
25%        1.000000   1190.000000   1951.000000      0.000000  98033.000000   
50%        1.500000   1560.000000   1975.000000      0.000000  98065.000000   
75%        2.000000   2210.000000   1997.000000      0.000000  98118.000000   
max        3.500000   9410.000000   2015.000000   2015.000000  98199.000000   

                lat          long  sqft_living15     sqft_lot15  
count  21597.000000  21597.000000   21597.000000   21597.000000  
mean      47.560093   -122.213982    1986.620318   12758.283512  
std        0.138552      0.140724     685.230472   27274.441950  
min       47.155900   -122.519000     399.000000     651.000000  
25%       47.471100   -122.328000    1490.000000    5100.000000  
50%       47.571800   -122.231000    1840.000000    7620.000000  
75%       47.678000   -122.125000    2360.000000   10083.000000  
max       47.777600   -121.315000    6210.000000  871200.000000

In [15]:
num_values = data['yr_renovated'].count()
num_values

17755

The dataset has 21597 entries and 20 columns. Most of the columns consist of numerical data which make it suitable for linear regression analysis.
The only columns with missing values is the year renovated column, hence we will drop the entries with missing values.